src/java/org/apache/cassandra/utils/memory/MemtablePool.java

```diff
@@ -44,26 +46,31 @@
     public final SubPool offHeap;
 
     public final Timer blockedOnAllocating;
+    public final Gauge<Long> numPendingTasks;
 
     final WaitQueue hasRoom = new WaitQueue();
 
-    MemtablePool(long maxOnHeapMemory, long maxOffHeapMemory, float cleanThreshold, Runnable cleaner)
+    MemtablePool(long maxOnHeapMemory, long maxOffHeapMemory, float cleanThreshold, MemtableCleaner cleaner)
     {
+        Preconditions.checkArgument(cleaner != null, "Cleaner should not be null");
+
         this.onHeap = getSubPool(maxOnHeapMemory, cleanThreshold);
         this.offHeap = getSubPool(maxOffHeapMemory, cleanThreshold);
         this.cleaner = getCleaner(cleaner);
-        blockedOnAllocating = CassandraMetricsRegistry.Metrics.timer(new DefaultNameFactory("MemtablePool")
-                                                                         .createMetricName("BlockedOnAllocation"));
-        if (this.cleaner != null)
-            this.cleaner.start();
+        this.blockedOnAllocating = CassandraMetricsRegistry.Metrics.timer(new DefaultNameFactory("MemtablePool")
+                                                                          .createMetricName("BlockedOnAllocation"));
+        this.cleaner.start();
+        this.numPendingTasks = CassandraMetricsRegistry.Metrics.register(new DefaultNameFactory("MemtableFlush")
+                                                                         .createMetricName("PendingTasks"),
+                                                                         () -> (long) this.cleaner.numPendingTasks());
```

Comment:
Once we have put the new `PendingTasks` metric task inside `MemtablePool` class we might also change the metric type from `MemtableFlush` to `MemtablePool`, and rename it to something like `pendingFlushTasks`. That way the two metrics (`BlockedOnAllocation` and `pendingFlushTasks`) would be grouped with a type matching their containing class:
```suggestion
        this.cleaner.start();
        DefaultNameFactory nameFactory = new DefaultNameFactory("MemtablePool");
        blockedOnAllocating = CassandraMetricsRegistry.Metrics.timer(nameFactory.createMetricName("BlockedOnAllocation"));
        numPendingTasks = CassandraMetricsRegistry.Metrics.register(nameFactory.createMetricName("PendingFlushTasks"),
                                                                    () -> (long) this.cleaner.numPendingTasks());
```